### PR TITLE
140/channels filter by connection

### LINKED ISSUE #140: Provide way to query all channels for a connection
Currently there is not way to determine which channel the relayer is processing. Would be great to see it in app.yaml. Also this could may help when migrating relayer process to another machine.

### PR DESCRIPTION
**Pull request by bartmacbartek**
Closes #140

### CODE REVIEW

File: src/binary/ibc-setup/commands/channels.spec.ts [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+test.before(async () => {
+  const ibcClientSimapp = await signingClient(simappChain, mnemonic);
+  const ibcClientWasm = await signingClient(wasmdChain, mnemonic);
+  link = await Link.createWithNewConnections(ibcClientWasm, ibcClientSimapp);
+  channel = await link.createChannel(
+    'A',
+    'transfer',
+    'custom',
+    Order.ORDER_UNORDERED,
+    'ics20-1'
```
Comment (willclarktech) [inline src/binary/ibc-setup/commands/channels.spec.ts:56]: Are these values not available from test utils?

File: src/binary/ibc-setup/commands/channels.spec.ts [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+    chain: 'local_wasm',
+    port: null,
+    connection: null,
+    mnemonic: null,
+    home: '/home/user',
+  };
+
+  fsReadFileSync.returns(registryYaml);
+
+  await run(options, (logger as unknown) as Logger);
```
Comment (willclarktech) [inline src/binary/ibc-setup/commands/channels.spec.ts:73]: Hm, maybe we should unify the `Logger` types in `lib` and `binary`.

File: src/binary/ibc-setup/commands/channels.spec.ts [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+
+  const output = consoleLog.getCall(-1).args[0] as string;
+
+  const channelWithDifferentPort = output
+    .split(os.EOL)
+    .slice(1) // remove table head
+    .find((value) =>
+      value.match(
+        new RegExp(
+          `[^\\s]+\\s+(?!${options.port})[^\\s]+\\s+[^\\s]+\\s+[^\\s]+`
```
Comment (willclarktech) [inline src/binary/ibc-setup/commands/channels.spec.ts:147]: Hm, this is a little hard to understand and seems easy to make a mistake and get false negatives here. How about checking that every channel has the correct port?

**Review by willclarktech — APPROVED** [review]
Looking good.

File: src/binary/ibc-setup/commands/channels.spec.ts (hunk 2 above)
Comment (bartmacbartek) [reply src/binary/ibc-setup/commands/channels.spec.ts:73]: Yes, I already spotted this here: https://github.com/confio/ts-relayer/pull/149#discussion_r608089104

Created a task: https://github.com/confio/ts-relayer/issues/155

File: src/binary/ibc-setup/commands/channels.spec.ts (hunk 3 above)
Comment (bartmacbartek) [reply src/binary/ibc-setup/commands/channels.spec.ts]: You're right. I inverted it and also spotted an edge case with last empty line.

I think it's easier to follow now, see: 281a400

File: src/binary/ibc-setup/commands/channels.spec.ts (hunk 1 above)
Comment (bartmacbartek) [reply src/binary/ibc-setup/commands/channels.spec.ts]: I didn't know but there's something. I had to flip things up but it works: https://github.com/confio/ts-relayer/pull/153/commits/c47a93696d3dcf62f3c4c403c3f77f7b9ca0f34a

**Pull request merged**